services/navigation/builtin/builtin_test.go

```diff
@@ -1011,3 +1010,116 @@ func TestValidateGeometry(t *testing.T) {
+		},
+		logger,
+	)
+	test.That(t, err, test.ShouldBeNil)
+
+	injectMovementSensor := inject.NewMovementSensor("test_movement")
+
+	injectMovementSensor.PositionFunc = func(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+		return geo.NewPoint(1, 1), 0, nil
+	}
+	injectMovementSensor.PropertiesFunc = func(ctx context.Context, extra map[string]interface{}) (*movementsensor.Properties, error) {
+		return &movementsensor.Properties{
+			CompassHeadingSupported: true,
+		}, nil
+	}
+	injectMovementSensor.CompassHeadingFunc = func(ctx context.Context, extra map[string]interface{}) (float64, error) {
+		// this is a left-handed value
+		return 50, nil
+	}
+
+	// create injected vis svc
+	injectedVis := inject.NewVisionService("test_vision")
+	injectedCam := inject.NewCamera("test_camera")
+
+	sphereGeom, err := spatialmath.NewSphere(spatialmath.NewZeroPose(), 1.0, "test-sphere")
+	test.That(t, err, test.ShouldBeNil)
+	sphereGob := spatialmath.NewGeoObstacle(geo.NewPoint(1, 1), []spatialmath.Geometry{sphereGeom})
+	gobCfg, err := spatialmath.NewGeoObstacleConfig(sphereGob)
+	test.That(t, err, test.ShouldBeNil)
+
+	ns, err := NewBuiltIn(
+		ctx,
+		resource.Dependencies{
+			injectMS.Name():             injectMS,
+			fakeBase.Name():             fakeBase,
+			injectMovementSensor.Name(): injectMovementSensor,
+			injectedVis.Name():          injectedVis,
+			injectedCam.Name():          injectedCam,
+		},
+		resource.Config{
+			ConvertedAttributes: &Config{
+				Store: navigation.StoreConfig{
+					Type: navigation.StoreTypeMemory,
+				},
+				BaseName:           "test_base",
+				MovementSensorName: "test_movement",
+				MotionServiceName:  "test_motion",
+				DegPerSec:          1,
+				MetersPerSec:       1,
+				MapType:            "",
+				Obstacles:          []*spatialmath.GeoObstacleConfig{gobCfg},
+				ObstacleDetectors: []*ObstacleDetectorNameConfig{
+					{VisionServiceName: injectedVis.Name().Name, CameraName: injectedCam.Name().Name},
+				},
+			},
+		},
+		logger,
+	)
+	test.That(t, err, test.ShouldBeNil)
+	defer func() {
+		test.That(t, ns.Close(context.Background()), test.ShouldBeNil)
+	}()
+
+	injectedVis.GetObjectPointCloudsFunc = func(ctx context.Context, cameraName string, extra map[string]interface{}) ([]*viz.Object, error) {
+		// this places the obstacle 60 degrees away from the base's local y-forward axis
+		// the obstacle is facing east, accounting got the current rotation of the base
+		boxGeom, err := spatialmath.NewBox(
+			spatialmath.NewPose(r3.Vector{math.Sqrt(3), 1, 0}, &spatialmath.OrientationVectorDegrees{OZ: 1, Theta: 320}),
+			r3.Vector{3.14, 2.72, 1},
+			"test-box",
+		)
+		test.That(t, err, test.ShouldBeNil)
+
+		detection, err := viz.NewObjectWithLabel(pointcloud.New(), "test-box", boxGeom.ToProtobuf())
+		test.That(t, err, test.ShouldBeNil)
+		return []*viz.Object{detection}, nil
+	}
+
+	manipulatedBoxGeom, err := spatialmath.NewBox(
+		spatialmath.NewPose(
```

Comment:
Isn't this saying that the box (which is of dimensions  3.14 mm, 2.72 mm, 1 mm) is at the origin facing east?

If so, I don't understand why the lat/lng wouldn't be `1,1` given that that box should be at the same lat/lng as the movement sensor is going to report that its lat/lng is `1,1`

Reply:
> Isn't this saying that the box (which is of dimensions 3.14 mm, 2.72 mm, 1 mm) is at the origin facing east?

I agree that the box has dimensions (3.14 mm, 2.72 mm, 1 mm) and is facing east.
However, I disagree that the box is at the origin. 
Below is `geometry` we declare in the injected `GetObjectPointClouds` func.
```
boxGeom, err := spatialmath.NewBox(
    spatialmath.NewPose(r3.Vector{math.Sqrt(3), 1, 0}, &spatialmath.OrientationVectorDegrees{OZ: 1, Theta: 320}),
    r3.Vector{3.14, 2.72, 1},
    "test-box",
)
```
Here, the geometry's pose as observed by the base is `math.Sqrt(3)` in the base's local X direction and `1` in the base's local Y direction.